engine/src/exec-interface-object.cpp

```diff
@@ -4704,3 +4707,126 @@ void MCObject::GetRevAvailableVariables(MCExecContext& ctxt, MCNameRef p_key, MC
     else
         r_variables = nil;
 }
+
+void MCObject::GetRevScriptDescription(MCExecContext& ctxt, MCValueRef& r_status)
+{
+    if (!getstack() -> iskeyed())
+    {
+        ctxt . LegacyThrow(EE_STACK_NOKEY);
+        return;
+    }
+    
+    bool t_success = true;
+    
+    MCAutoValueRefBase<MCScriptObjectRef> t_object_ref;
+    
+    if (t_success)
+        t_success = MCEngineScriptObjectCreate(this, 0, &t_object_ref);
+    
+    if (t_success)
+        r_status = MCEngineExecDescribeScriptOfScriptObject(*t_object_ref);
+    
+    if (t_success &&
+        !MCExtensionConvertToScriptType(ctxt, r_status))
+    {
+        t_success = false;
+    }
+}
+
+void MCObject::GetEffectiveRevScriptDescription(MCExecContext& ctxt, MCValueRef& r_descriptions)
+{
+    bool t_success;
+    t_success = true;
+    
+    MCAutoArrayRef t_descriptions;
+    t_success = MCArrayCreateMutable(&t_descriptions);
+    
+    MCObjectList *t_object_list;
+    t_object_list = nil;
+    
+    t_success = MCObjectListAppendObjectList(t_object_list, MCfrontscripts);
+    
+    for (MCObject *t_object = this; t_success && t_object != NULL; t_object = t_object -> parent)
+    {
+        t_success = MCObjectListAppendObjectAndBehaviors(t_object_list, t_object);
+    }
+    
+    if (t_success)
+        t_success = MCObjectListAppendObjectList(t_object_list, MCbackscripts);
+    
+    for (uint32_t i = 0; t_success && i < MCnusing; i++)
+    {
+        if (MCusing[i] == this)
+            continue;
+        t_success = MCObjectListAppendObjectAndBehaviors(t_object_list, MCusing[i]);
+    }
+    
+    if (t_success)
+    {
+        MCObjectList *t_object_ref;
+        t_object_ref = t_object_list;
+        
+        index_t t_index = 1;
+        do
+        {
+            if (!t_object_ref -> getremoved() && t_object_ref -> getobject() -> getstack() -> iskeyed())
+            {
+                MCAutoValueRef t_description;
+                
+                MCAutoValueRefBase<MCScriptObjectRef> t_script_object_ref;
+                
+                if (t_success)
+                    t_success = MCEngineScriptObjectCreate(t_object_ref -> getobject(), 0, &t_script_object_ref);
+                
+                if (t_success)
+                    t_description = MCEngineExecDescribeScriptOfScriptObject(*t_script_object_ref, t_object_ref -> getobject() == this);
```

Comment:
If `t_success` is false at some point here I guess we want to just exit the loop?